tests/slurm_plugin/test_capacity_block_manager.py

```diff
@@ -0,0 +1,745 @@
+            if expected_internal_error:
+                assert_that(caplog.text).contains("Unable to find Capacity Block")
+                assert_that(capacity_block_manager._reserved_nodenames).is_equal_to(expected_reserved_nodenames)
+            else:
+                assert_that(reserved_nodenames).is_equal_to(expected_reserved_nodenames)
+
+        if is_time_to_update and not expected_ec2_exception:
+            if all(not action_succeeded for action_succeeded in slurm_reservation_creation_succeeded):
+                # if all slurm reservations actions failed, do not update the values
+                assert_that(capacity_block_manager._capacity_blocks_update_time).is_equal_to(previous_update_time)
+                assert_that(capacity_block_manager._capacity_blocks).is_not_equal_to(capacity_blocks_from_config)
+            else:
+                assert_that(capacity_block_manager._capacity_blocks_update_time).is_not_equal_to(previous_update_time)
+                assert_that(capacity_block_manager._capacity_blocks).is_equal_to(capacity_blocks_from_config)
+
+            update_res_mock.assert_has_calls(
+                [
+                    call(capacity_block=capacity_block, do_update=False)
+                    for capacity_block in capacity_block_manager._capacity_blocks.values()
+                ],
+                any_order=True,
+            )
+            cleanup_mock.assert_called_once()
+        else:
+            assert_that(capacity_block_manager._capacity_blocks_update_time).is_equal_to(previous_update_time)
+            update_res_mock.assert_not_called()
+            cleanup_mock.assert_not_called()
+
+    @pytest.mark.parametrize(
+        ("is_initialized", "now", "expected_updated_time"),
+        [
+            # manager not initialized
+            (False, datetime(2020, 1, 1, 1, 00, 0), True),
+            # delta < CAPACITY_BLOCK_RESERVATION_UPDATE_PERIOD
+            (True, datetime(2020, 1, 1, 1, 00, 10), False),
+            (True, datetime(2020, 1, 1, 1, 9, 0), False),
+            # delta >= CAPACITY_BLOCK_RESERVATION_UPDATE_PERIOD
+            (True, datetime(2020, 1, 1, 1, 10, 0), True),
+            (True, datetime(2020, 1, 1, 1, 21, 0), True),
+            (True, datetime(2020, 1, 1, 2, 00, 0), True),
+            (True, datetime(2020, 1, 2, 1, 00, 0), True),
+        ],
+    )
+    def test_is_time_to_update_capacity_blocks_info(
+        self, capacity_block_manager, is_initialized, now, expected_updated_time
+    ):
+        capacity_block_manager._capacity_blocks_update_time = datetime(2020, 1, 1, 1, 00, 0) if is_initialized else None
+        assert_that(capacity_block_manager._is_time_to_update_capacity_blocks_info(now))
+
+    @pytest.mark.parametrize(
+        ("capacity_blocks", "nodes", "expected_nodenames_in_capacity_block"),
+        [
+            (
+                {
+                    "cr-123456": CapacityBlock("id", "queue-cb", "compute-resource-cb"),
```

Comment:
* please add test case with no capacity blocks.
* can the same reservation ID be set on two (or more) different queues? How is this modeled? I suppose by two different CapacityBlock, please add the test case

Reply:
This is a nice catch, thanks! 

Added a commit: https://github.com/aws/aws-parallelcluster-node/pull/591/commits/2a687fc9cad99b70d7e3932e8e978208fe31e9f6 to add support for multiple queues/compute-resources for the same CB.